services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt

```diff
@@ -80,23 +75,9 @@ class HankeController(@Autowired private val hankeService: HankeService) {
         }
     }
 
-    /**
-     * Get all hanke
-     *  TODO: token  from front?
-     *  TODO: limit call with user information and return only user's own hanke or something?
-     *  TODO: do we later on have users who can read all Hanke items?
-     */
-    internal fun getAllHankeItems(): ResponseEntity<Any> {
-        logger.info { "Entering getAllHankeItems" }
-        return try {
-            val hankeList = hankeService.loadAllHanke()
-            ResponseEntity.status(HttpStatus.OK).body(hankeList)
-
-        } catch (e: Exception) {
-            logger.error(e) {
-                HankeError.HAI1004.toString()
-            }
-            ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(HankeError.HAI1004)
+    private fun includeGeometry(hankeList: List<Hanke>) {
+        hankeList.forEach { hanke ->
+            hanke.geometriat = hankeGeometriatService.loadGeometriat(hanke.hankeTunnus!!)
```

Comment:
Did you consider getting all geometrys in one call? Now service is called separately for all hanke. If this is thought process then OK :)

Reply:
I did not consider changing into that. But I did change what you suggested elsewhere about loading geometries without also loading the Hanke first (as was done for the geometry loading with only hankeTunnus before).
Loading geometries for multiple Hanke at once would need more changes into database queries etc. Something for future development perhaps...